ext/readline/readline.c

```diff
@@ -187,8 +187,10 @@ PHP_FUNCTION(readline_info)
 					RETURN_THROWS();
 				}
 #ifndef PHP_WIN32
-				if (strlen(oldstr) < Z_STRLEN_P(value)) {
+				if (!oldstr || strlen(oldstr) < Z_STRLEN_P(value)) {
 					rl_extend_line_buffer(Z_STRLEN_P(value) + 1);
+				} else if (!rl_line_buffer) {
+					rl_line_buffer = malloc(Z_STRLEN_P(value) + 1);
 				}
```

Comment:
I am confused, surely having the `if (!rl_line_buffer)` check first makes more sense? As you then don't need to check for `!oldstr` ?